### DISCUSSION TITLE
Parent Child devices in rack

### DISCUSSION THREAD

**Discussion by jadams3710** [netbox-community/netbox#5437] [Q&A] [answered]
We have devices that are set up as Parent devices, with child devices being modules inserted into the parent device.
The child device still shows as unracked even though it is in the parent device which is mounted.  Is there a different way to set these up so that devices show in the rack?
One example is we have a network device with 4 bays one bays is taken by the management controller, 2 bays by 48 port cat6 ethernet switch modules, and last module taken by fiber module.
All the modules show non-racked even though they are in the rack. Just trying to show true representation of the rack so when we give information to the data centers to help us we know what is truly in our rack and cabling diagrams etc... 
Didn't know if this was was know issue or future plans.
Thanks for the application it is a great tool to have.

**Comment by candlerb** [comment, marked as answer]
> One example is we have a network device with 4 bays one bays is taken by the management controller, 2 bays by 48 port cat6 ethernet switch modules, and last module taken by fiber module.

That's not how device bays are intended to be used in Netbox.  "Device bays" are for independently managed devices which sit in an outer, rack-mounted housing - most commonly blade servers in a shared chassis.  Apart from sharing power, they are independent devices.

For your use case, you should model the network device as a regular Device, and the management controller and ethernet modules as Inventory Items.  Create the ethernet ports directly on the Device.  (Currently it's not possible to put ports on an inventory item)

Now, there is a separate issue:

> The child device still shows as unracked even though it is in the parent device which is mounted. Is there a different way to set these up so that devices show in the rack?

The issue boils down to: "if I have a chassis containing blade servers, should those blade servers show as 'racked' rather than 'unracked'?"  In one sense ideally they should, but the rendering that Netbox provides isn't capable of this, as it basically just renders a rectangle for the chassis itself.  In another sense they shouldn't: the blade *chassis* is mounted in the rack, but the blade servers themselves are not directly mounted in the rack - that is, they aren't 19" wide and don't screw into the rack rails.

**Reply by jadams3710** [reply]
Thanks for the clarification that does make sense, 
We will just keep them as device bays because we also have shelves that have several different devices sitting on them and we need to show interfaces since they connect to patch panels and switches.

**Comment by AnythingOverIP** [comment]
We model the DRAC and ILO card of servers as a device bay component. They are part of a racked server but they have it's own OS, hostname and everything. In this case, I would prefer not to have it shown as a non-racked. When generating the "non-racked" table, having a simple check to see if child is attached to a racked device in order to filter it would be nice.
